packages/client/src/pages/RegisterAndLogin/RegisterAndLogin.tsx

```diff
@@ -0,0 +1,100 @@
+import { Box, useTheme } from '@mui/material';
+
+import dish from '../../assets/dish.jpg';
+import restaurant from '../../assets/restaurant.jpg';
+import { LoginForm, Question, RegisterForm, WelcomeMessage } from '../../components';
+import { routes } from '../../config';
+
+const userLogin = () => (
+  <>
+    <WelcomeMessage message='Witamy ponownie!' />
+    <LoginForm />
+    <Question question='Nie masz konta?' link={routes.userRegister} linkText='Zarejestruj się' />
+  </>
+);
+
+const partnerLogin = () => (
+  <>
+    <WelcomeMessage message='Witamy ponownie!' />
+    <LoginForm />
+    <Question question='Chcesz do nas dołączyć?' link={routes.partnerRegister} linkText='Zarejestruj się' />
+  </>
+);
+
+const userRegister = () => (
+  <>
+    <WelcomeMessage message='Stwórz konto' />
+    <RegisterForm userRole='User' />
+    <Question question='Masz już konto?' link={routes.userLogin} linkText='Zaloguj się' />
+  </>
+);
+
+const partnerRegister = () => (
+  <>
+    <WelcomeMessage message='Dołącz do naszych partnerów' />
+    <RegisterForm userRole='Partner' />
+    <Question question='Jesteś już w naszym programie?' link={routes.partnerLogin} linkText='Zaloguj się' />
+  </>
+);
+
+type RegisterAndLoginParams = {
+  formType: 'UserLogin' | 'UserRegister' | 'PartnerLogin' | 'PartnerRegister';
+};
+
+const RegisterAndLogin = ({ formType }: RegisterAndLoginParams) => {
+  const theme = useTheme();
+
+  return (
+    <Box
+      sx={{
+        minHeight: 'calc(100vh - 6rem)',
+        backgroundColor: theme.palette.background.default,
+        backgroundImage:
+          formType === 'UserLogin' || formType === 'UserRegister' ? `url(${dish})` : `url(${restaurant})`,
+        backgroundSize: `cover`,
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
+    >
+      <Box
+        sx={{
+          width: '45rem',
+          margin: '1rem',
+          background: theme.palette.common.white,
```

Comment:
```suggestion
          background: theme.palette.background.default,
```
Wszystko działa i wygląda w kodziesuper ! Tylko zauważyłam, że tam jest problem przy zmianie `mode`. Użyłaś `common.white` a on jest właśnie taki sam niezależnie od `mode`. Tu Ci proponuje nasz defaultowy kolor, to nie jest biały ale też pasuje i to najszybsza zmiana. 

Ew.  druga opcja to nadpisać w 'dark mode' ten kolor 'common.white' ale chyba nie ma sensu, bo z nazwy wnioskuje że to ma być wspólny.